operator/src/test/java/org/keycloak/operator/PodTemplateE2EIT.java

```diff
@@ -0,0 +1,175 @@
+import static org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition.HAS_ERRORS;
+
+@QuarkusTest
+public class PodTemplateE2EIT extends ClusterOperatorTest {
+
+    private Keycloak getEmptyPodTemplateKeycloak() {
+        return Serialization.unmarshal(getClass().getResourceAsStream("/empty-podtemplate-keycloak.yml"), Keycloak.class);
+    }
+
+    @Test
+    public void testPodTemplateIsMerged() {
+        // Arrange
+        var keycloakWithPodTemplate = k8sclient
+                .load(getClass().getResourceAsStream("/correct-podtemplate-keycloak.yml"));
+
+        // Act
+        keycloakWithPodTemplate.createOrReplace();
+
+        // Assert
+        Awaitility
+                .await()
+                .ignoreExceptions()
+                .atMost(3, MINUTES).untilAsserted(() -> {
+            Log.info("Getting logs from Keycloak");
+
+            var keycloakPod = k8sclient
+                    .pods()
+                    .inNamespace(namespace)
+                    .withLabel("app", "keycloak")
+                    .list()
+                    .getItems()
+                    .get(0);
+
+            var logs = k8sclient
+                    .pods()
+                    .inNamespace(namespace)
+                    .withName(keycloakPod.getMetadata().getName())
+                    .getLog();
+
+            Log.info("Full logs are:\n" + logs);
+            assertThat(logs).contains("Hello World");
+            assertThat(keycloakPod.getMetadata().getLabels().get("foo")).isEqualTo("bar");
+        });
+    }
+
+    @Test
+    public void testPodTemplateIncorrectName() {
+        // Arrange
+        var plainKc = getEmptyPodTemplateKeycloak();
+        var podTemplate = new PodTemplateSpecBuilder()
+                .withNewMetadata()
+                .withName("foo")
+                .endMetadata()
+                .build();
+        plainKc.getSpec().getUnsupported().setPodTeplate(podTemplate);
+
+        // Act
+        k8sclient.resource(plainKc).createOrReplace();
+
+        // Assert
+        Log.info("Getting status of Keycloak");
+        Awaitility
+                .await()
+                .ignoreExceptions()
+                .atMost(3, MINUTES).untilAsserted(() -> {
+                    var message = k8sclient
+                            .resources(Keycloak.class)
+                            .inNamespace(namespace)
+                            .withName("example-podtemplate")
+                            .get()
+                            .getStatus()
+                            .getConditions()
+                            .stream()
+                            .filter(c -> c.getType().equals(HAS_ERRORS))
+                            .findFirst()
+                            .get()
+                            .getMessage();
+
+                    Log.info("Resource error message is:" + message);
+                    assertThat(message).contains("cannot be modified");
```

Comment:
Maybe we could create a re-usable util method for this, similar to `assertKeycloakStatusCondition`? This might be useful for other tests too.

Reply:
changed `assertKeycloakStatusCondition` to return the message of the matched condition.

Reply:
done